hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -290,7 +291,7 @@ public SecurityContext getSecurityContext() {
 
     @Override
     public boolean bind(final ClientEndpoint endpoint) {
-        if (!clientSelector.select(endpoint)) {
+        if (isMCClient(endpoint) && !clientSelector.select(endpoint)) {
```

Comment:
I think you meant to do 
`
if (!isMCClient(endpoint) && !clientSelector.select(endpoint))
`

Reply:
Yes, I've overlooked this one. Good catch!